server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -1101,10 +1101,7 @@ public void writeOptionalStringCollection(final Collection<String> collection) t
      * Writes a list of {@link NamedWriteable} objects.
      */
     public void writeNamedWriteableList(List<? extends NamedWriteable> list) throws IOException {
-        writeVInt(list.size());
-        for (NamedWriteable obj : list) {
-            writeNamedWriteable(obj);
-        }
+        writeCollection(list, StreamOutput::writeNamedWriteable);
```

Comment:
This one I also found :P https://github.com/elastic/elasticsearch/pull/99087

Reply:
Heh, I'll back this out to avoid colliding with yours.